19 create login view

aviel-lanzman opened the pull request:
in client:

1. create login and register page
2. add pinia auth
3. add pre commit hook for force format by prettier
4. add logo

in server:

1. fix the parser body in auth handler
2. add sending user parameters isSuperuser
3. add cors
4. add cookie expires 24 hour
5. fix in user model before create hook

aviel-lanzman committed:
add logo

aviel-lanzman committed:
add .prettierrc to client

aviel-lanzman committed:
add link to login

aviel-lanzman committed:
add app bar query if show it

aviel-lanzman committed:
add cors  config

aviel-lanzman committed:
add vscode debug server config

aviel-lanzman committed:
fix before create hook

aviel-lanzman committed:
fix body parser with user model

aviel-lanzman committed:
add process.env

aviel-lanzman committed:
add pre commit formatter for client

aviel-lanzman committed:
add login page

aviel-lanzman committed:
fix: login handler fix body parser error

aviel-lanzman committed:
add: expires for cookie 24 hour

aviel-lanzman committed:
add: AllowCredentials for enable cookies

aviel-lanzman committed:
add: send to client isSuperuser

aviel-lanzman committed:
add: auth pinia store

aviel-lanzman committed:
add: check login in create app

aviel-lanzman committed:
add: login route

aviel-lanzman committed:
add: axios

aviel-lanzman committed:
add: webFontLoader

aviel-lanzman committed:
add: Error handler

Bokoness commented:
Thanks!
@aviel-lanzman any chance that you can play with the prettier? It's really hard to tell which changes are related to the PR. Please make sure that you are using the same formatting rules as it was, and that there are no formatting changes in the PR.

aviel-lanzman commented:
Gladly 😀
What prettier rules do you use?
I added a rules file so that these problems don't exist, and I also added a script pre-commit that will enforcement the rules of the prettier.

Bokoness commented:
@aviel-lanzman If you wanna do that, open a separate PR for it, right now, just make sure that only the relevant changes are in this PR

aviel-lanzman committed:
fix: pr details

aviel-lanzman committed:
fix: pr details

aviel-lanzman committed:
test

aviel-lanzman commented:
@Bokoness  
I fixed it
I think

aviel-lanzman committed:
add: if user is logged in, replace the login button with the user's avatar

aviel-lanzman committed:
add: throw error from login and register stor

aviel-lanzman committed:
fix: posh to home apter login
